sql/expression/function/uuid_test.go

```diff
@@ -0,0 +1,211 @@
+// Copyright 2021 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package function
+
+import (
+	"testing"
+
+	"github.com/dolthub/vitess/go/vt/proto/query"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/require"
+
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/expression"
+)
+
+func TestUUID(t *testing.T) {
+	// Generate a UUID and validate that is a legitimate uuid
+	uuidE := NewUUIDFunc()
+	ctx := sql.NewEmptyContext()
+
+	result, err := uuidE.Eval(ctx, sql.Row{nil})
+	require.NoError(t, err)
+
+	myUUID := result.(string)
+	_, err = uuid.Parse(myUUID)
+	require.NoError(t, err)
+
+	// validate that generated uuid is legitimate for IsUUID
+	val := NewIsUUID(uuidE)
+	require.Equal(t, int8(1), eval(t, val, sql.Row{nil}))
```

Comment:
May want to throw a regex in here as well, since we can't verify that the `uuid` library doesn't accept invalid UUIDs or UUIDs in an invalid form (such as an arbitrary number of dashes).

Reply:
Oh I see. That's a good idea.